airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/DatabaseConfigPersistence.java

```diff
@@ -1575,17 +1583,47 @@ public void replaceAllConfigs(final Map<AirbyteConfig, Stream<?>> configs, final
       result.put(ConfigSchema.STANDARD_SOURCE_DEFINITION.name(),
           standardSourceDefinitionWithMetadata
               .stream()
-              .map(ConfigWithMetadata::getConfig)
-              .map(Jsons::jsonNode));
+              .map(configWitMetadata -> {
+                if (!featureFlags.hideSecretsInExport()) {
+                  return Jsons.jsonNode(configWitMetadata.getConfig());
+                }
+
+                try {
+                  final StandardSourceDefinition standardSourceDefinition = getConfig(
+                      ConfigSchema.STANDARD_SOURCE_DEFINITION,
+                      configWitMetadata.getConfigId(),
+                      StandardSourceDefinition.class);
+                  final JsonNode connectionSpecs = standardSourceDefinition.getSpec().getConnectionSpecification();
+                  final JsonNode sanitizedConfig = jsonSecretsProcessor.maskSecrets(Jsons.jsonNode(configWitMetadata.getConfig()), connectionSpecs);
+                  return sanitizedConfig;
+                } catch (final ConfigNotFoundException | JsonValidationException | IOException e) {
+                  throw new RuntimeException(e);
+                }
+              }));
     }
     final List<ConfigWithMetadata<StandardDestinationDefinition>> standardDestinationDefinitionWithMetadata =
         listStandardDestinationDefinitionWithMetadata();
     if (!standardDestinationDefinitionWithMetadata.isEmpty()) {
       result.put(ConfigSchema.STANDARD_DESTINATION_DEFINITION.name(),
           standardDestinationDefinitionWithMetadata
               .stream()
-              .map(ConfigWithMetadata::getConfig)
-              .map(Jsons::jsonNode));
+              .map(configWitMetadata -> {
+                if (!featureFlags.hideSecretsInExport()) {
+                  return Jsons.jsonNode(configWitMetadata.getConfig());
+                }
+
+                try {
+                  final StandardDestinationDefinition standardDestinationDefinition = getConfig(
+                      ConfigSchema.STANDARD_DESTINATION_DEFINITION,
+                      configWitMetadata.getConfigId(),
+                      StandardDestinationDefinition.class);
+                  final JsonNode connectionSpecs = standardDestinationDefinition.getSpec().getConnectionSpecification();
+                  final JsonNode sanitizedConfig = jsonSecretsProcessor.maskSecrets(Jsons.jsonNode(configWitMetadata.getConfig()), connectionSpecs);
```

Comment:
is this right? maybe i'm confused but this looks like it is sanitizing the destination definition not the destination itself. am i missing something?

Reply:
Same comment than the Source part. I have changed it.